channeldb/payments.go

```diff
@@ -525,10 +528,18 @@ func serializeHop(w io.Writer, h *route.Hop) error {
 		return WriteElements(w, uint32(0))
 	}
 
+	// Gather all non-primitive TLV records so that they can be serialized
+	// as a single blob.
+	var records []tlv.Record
+	if h.MPP != nil {
+		records = append(records, h.MPP.Record())
+	}
+	records = append(records, h.TLVRecords...)
+
 	// Otherwise, we'll transform our slice of records into a map of the
 	// raw bytes, then serialize them in-line with a length (number of
 	// elements) prefix.
-	mapRecords, err := tlv.RecordsToMap(h.TLVRecords)
+	mapRecords, err := tlv.RecordsToMap(records)
```

Comment:
indeed I agree, I have raised this concern several times in the past but so far the concerns have been dismissed every time. this is one of the reasons I pushed for disabling custom records in 0.8 until this had some resolution. 

my preferred solution is to store everything in a single blob, then the known fields can be extracted directly into the struct, and unknown ones can be extracted into their own map. 

the downside to writing the fields in the traditional way is that we need to add migrations to place the new fields in their logical spot, e.g after existing fields but before the custom records. this is why I prefer serializing everything in one blob, but never we never reached agreement that this was the way to proceed. it will also require a migration at this point since the current split has already been deployed..

Reply:
Is it a problem if they new fields aren't in a logical spot? We can also keep adding all new known fields at the end.

Reply:
squashing down